library/build.gradle

```diff
@@ -15,32 +15,92 @@ apply plugin: 'signing'
 archivesBaseName = 'android-maps-utils'
 group = 'com.google.maps.android'
 
-dependencies {
-    compile 'com.google.android.gms:play-services-maps:6.5.87'
-}
-
-
 android {
     compileSdkVersion 19
     buildToolsVersion "21.1.2"
+    defaultPublishConfig "release"
+    publishNonDefault true
 
     sourceSets {
         main {
             manifest.srcFile 'AndroidManifest.xml'
+        }
+
+        gms{
             java.srcDirs = ['src']
             res.srcDirs = ['res']
         }
 
+        m4b {
+            java.srcDirs = ['m4b/src']
+            res.srcDirs = ['m4b/res']
+        }
+
         androidTest {
             java.srcDirs = ['tests/src']
         }
     }
+
+    productFlavors {
+        gms {
+            dependencies {
+                compile 'com.google.android.gms:play-services-maps:6.5.87'
+            }
+        }
+
+        m4b {
+            dependencies {
```

Comment:
I suppose its "applicationId" declaration. I tried applicationId, but I got the following error. It looks that the android-library can't have its applicationId since it is not an application.

Error:Library projects cannot set applicationId. applicationId is set to 'com.google.maps.android.m4b' in flavor 'm4b'.